examples/train_NGCF.py

```diff
@@ -0,0 +1,105 @@
+import sys
+from beta_rec.datasets.dunnhumby import Dunnhumby
+sys.path.append("../")
+import os
+import json
+import numpy as np
+import argparse
+import torch
+import pandas as pd
+from datetime import datetime
+from tqdm import tqdm
+from beta_rec.models.NGCF import NGCF
+from beta_rec.models.NGCF import NGCFEngine
+from beta_rec.datasets.NGCF_data_utils import Data
+from beta_rec.datasets import dataset
+from beta_rec.datasets.movielens import Movielens_100k
+
+
+def parse_args():
+    """
+    Parse args from command line
+    Returns:
+
+    """
+    parser = argparse.ArgumentParser(description="Run NGCF..")
+    parser.add_argument(
+        "--config_file",
+        nargs="?",
+        type=str,
+        default="../configs/NGCF_default.json",
+        help="Specify the config file name. Only accept a file from ../configs/",
+    )
+
+    parser.add_argument(
+        "--emb_dim", nargs="?", type=int, help="Dimension of the embedding."
+    )
+    parser.add_argument("--lr", nargs="?", type=float, help="Initialize learning rate.")
+    parser.add_argument("--num_epoch", nargs="?", type=int, help="Number of max epoch.")
+
+    parser.add_argument(
+        "--batch_size", nargs="?", type=int, help="Batch size for training."
+    )
+    return parser.parse_args()
+
+def update_args(config, args):
+    """Update config parameters by the received parameters from command line
+
+        Args:
+            config (dict): Initial dict of the parameters from JOSN config file.
+            args (object): An argparse Argument object with attributes being the parameters to be updated.
+
+        Returns:
+            None
+    """
+    for k, v in vars(args).items():
+        if v != None:
+            config[k] = v
+            print("Received parameters form comand line:", k, v)
+
+
+def sparse_mx_to_torch_sparse_tensor(sparse_mx):
+    """Convert a scipy sparse matrix to a torch sparse tensor."""
```

Comment:
Please improve the code comments using 
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html